source/common/stats/thread_local_store.cc

```diff
@@ -194,11 +197,25 @@ void ThreadLocalStoreImpl::initializeThreading(Event::Dispatcher& main_thread_di
   tls_cache_ = ThreadLocal::TypedSlot<TlsCache>::makeUnique(tls);
   tls_cache_->set(
       [](Event::Dispatcher&) -> std::shared_ptr<TlsCache> { return std::make_shared<TlsCache>(); });
+  tls_ = tls;
 }
 
 void ThreadLocalStoreImpl::shutdownThreading() {
   // This will block both future cache fills as well as cache flushes.
   shutting_down_ = true;
+
+  if (tls_.has_value()) {
+    ASSERT(tls_->isShutdown());
+  }
+
+  // We can't call runOnAllThreads here as global threading has already been shutdown.
+  // It is okay to simply clear the scopes and central cache entries to cleanup.
```

Comment:
Can you include an explanation for why this is okay to do?

Reply:
Will update. How is:
`  // We can't call runOnAllThreads here as global threading has already been shutdown. It is okay
  // to simply clear the scopes and central cache entries here as they will be cleaned up during thread local data
  // cleanup in InstanceImpl::shutdownThread().`